file-attribute-caching/src/main/kotlin/com/pkware/filesystem/attributecaching/AttributeCachingPath.kt

```diff
@@ -162,22 +219,37 @@ internal class AttributeCachingPath(
     override fun toString(): String = delegate.toString()
 
     /**
-     * Sets the entry for the given attribute [name] with the given [value]. Can only set entire
-     * attribute `Class`es such as "dos:*", "posix:*", and "basic:*"
+     * Sets the entry for the given attribute [name] with the given [value]. Can only set entire attribute `Class`es
+     * such as "acl:*", "dos:*", "posix:*", and "basic:*" For most attribute `Class`es the actual attributes are cached,
+     * but in the case of [AclFileAttributeView] the entire view is cached.
      *
      * The attribute name must include a "*" in order to be set within the cache.
      *
      * @param name The name of the attribute to cache.
-     * @param value The attribute value to cache.
+     * @param value The [FileAttributeView] to cache from.
      */
-    fun <A : BasicFileAttributes> setAttributeByName(name: String, value: A?) {
+    fun <A : FileAttributeView> setAttributeByName(name: String, value: A?) {
         // This check is to ensure that we are only storing attribute classes and not specific attributes.
         // Remove basic: from our attribute name if present as basicFileAttributes can be accessed without that
         // qualifier
         when (name.substringAfter("basic:")) {
```

Comment:
I still wish Java had something like an attribute parsing component for this or ours was a little smarter.

We technically allow input like `basic:posix:group` or `basic:acl:owner`, returning the properties without error. It's harmless but strange.